### PR TITLE
Fix a bug with `num_keys` in `_scatter_by_slice`

### PR DESCRIPTION
**Pull request by thomcom**
## Description
This PR closes https://github.com/rapidsai/cudf/issues/12748 by changing the `num_keys` computation in `column._scatter_by_slice`.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/core/column/column.py [hunk 1]
```diff
@@ -570,7 +571,7 @@ def _scatter_by_slice(
         start, stop, step = key.indices(len(self))
         if start >= stop:
             return None
-        num_keys = (stop - start) // step
+        num_keys = ceil((stop - start) / step)
```
Comment (bdice) [inline python/cudf/cudf/core/column/column.py:574]: Should we worry about floating precision here? We can do this entirely as an integral computation.
```suggestion
        num_keys = (stop - start) // step)
        if (stop - start) % step:
            num_keys += 1
```

or even `divmod`, which I _think_ uses only integral math:
```suggestion
        num_keys, remainder = divmod(stop - start, step)
        if remainder:
            num_keys += 1
```

Comment (wence-) [reply python/cudf/cudf/core/column/column.py:574]: How about: `len(range(len(self))[key])`

Comment (thomcom) [reply python/cudf/cudf/core/column/column.py:574]: I'm not sure what the danger is w.r.t. floating point precision. Isn't the worst case going to be ((2^^32 - (2^^32-1)) / (2^^32-1)). `ceil` works fine on this value, anyway.

I'm not a fan of `len(range(len(self))[key]`. That has us building python range objects that are as much as 2^^32 in size, major performance crush.

Comment (bdice) [reply python/cudf/cudf/core/column/column.py:574]: @thomcom Indexing a `range` with a `slice` produces another `range`. It won't materialize that as a list of the indices.

Comment (bdice) [reply python/cudf/cudf/core/column/column.py:574]: @wence- has the best solution here afaict. Good call. I should've zoomed out and seen that in context. :)
```suggestion
        num_keys = len(range(len(self))[key])
```

Comment (wence-) [reply python/cudf/cudf/core/column/column.py:574]: Probably it is more obviously written as:
```
len(range(start, stop, step))
```

**Comment by thomcom** [comment]
Did that

**Comment by thomcom** [comment]
rerun tests

**Comment by thomcom** [comment]
This is ready to merge I think, and a PR in cuSpatial depends on it (https://github.com/rapidsai/cuspatial/pull/926)

**Review by wence- — APPROVED** [review]
Sorry, I thought I had hit approve, but no. Looks great, thanks!

**Pull request merged**